packages/starlight/utils/navigation.ts

```diff
@@ -98,10 +98,16 @@ function groupFromAutogenerateConfig(
 /** Check if a string starts with one of `http://` or `https://`. */
 const isAbsolute = (link: string) => /^https?:\/\//.test(link);
 
+/** Ensure the passed path ends with a trailing slash. */
+function ensureTrailingSlash(href: string): string {
+	if (href[href.length - 1] !== '/') href += '/';
+	return href;
+}
```

Comment:
Might it make sense to centralise these helpers? We also have these `strip` methods in our base utils:
https://github.com/withastro/starlight/blob/c344ee28b11dfeb951af3da21f3332ac7b202568/packages/starlight/utils/base.ts#L15-L20
If the non RegExp version is that much faster, we can refactor these.

Reply:
Good catch, only saw the ones in the same file. I think we might as well do that now and be done with it while we're at it and it does not even need a changeset.

This would means we would have:

- `ensureLeadingSlash()`
- `ensureTrailingSlash()`
- `ensureLeadingAndTrailingSlashes()` _(using `ensureLeadingSlash()` and `ensureTrailingSlash()`)_
- `stripLeadingSlash()`
- `stripTrailingSlash()`

Now the big question ^^ Do we want to centralize them in `utils/navigation.ts`, `utils/base.ts` or a new file e.g. `utils/path.ts`, `utils/string.ts`?

For the regexp, I'll check with a new benchmark for stripping but like I said, I am not sure it's even properly benchmarked.

Reply:
It probably doesn’t matter _hugely_ where these end up. I’d say they deserve a dedicated file, but don’t mind too much about naming!

I wouldn’t be surprised if the RegExp version is slower than the simple index lookups — after all they should require compiling and running the pattern vs a quick reference in the character array.